### PR TITLE
sql: hashes with NULL input should return NULL

### LINKED ISSUE #29754: sql: md5(null) should return null
And probably a buncha other similar functions.

### PR DESCRIPTION
**Pull request by rjnn**
Fixes #29754.

### CODE REVIEW

File: pkg/sql/sem/builtins/builtins.go [hunk 1]
```diff
@@ -3566,7 +3571,9 @@ func hashBuiltin(newHash func() hash.Hash, info string) builtinDefinition {
 			ReturnType: tree.FixedReturnType(types.String),
 			Fn: func(_ *tree.EvalContext, args tree.Datums) (tree.Datum, error) {
 				h := newHash()
-				feedHash(h, args)
+				if feedHash(h, args) == 0 {
```
Comment (mjibson) [inline pkg/sql/sem/builtins/builtins.go]: is `md5('')`? tested anywhere? gotta make sure that still works.

Comment (arjunravinarayan) [reply pkg/sql/sem/builtins/builtins.go]: Oh yeah, hmm, I broke that. Let me rework. Thanks.

**Review by rjnn — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/29822)* status: :shipit: complete! 0 of 0 LGTMs obtained

___
*[pkg/sql/sem/builtins/builtins.go, line 3574 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/29822#-LLpls2VcM9jJ59SjSdQ-r1-3574:-LLq5jv48PJ4O2ZqRaZU:b6d66we) ([raw file](https://github.com/cockroachdb/cockroach/blob/7c9e5bda80b045b37946f8447f7aa2c2eee79d5f/pkg/sql/sem/builtins/builtins.go#L3574)):*
<details><summary><i>Previously, arjunravinarayan (Arjun Narayan) wrote…</i></summary><blockquote>

Oh yeah, hmm, I broke that. Let me rework. Thanks.
</blockquote></details>

Fixed, and added a logic test. PTAL.

**Review by madelynnblue — APPROVED** [review]

**Comment by arjunravinarayan** [comment]
bors r=mjibson

**Pull request merged**